### PR TITLE
Refactor analyses

### PR DESCRIPTION
**Pull request by henrikstranneheim**
### Added

- Analysis filters for `is_visible` and `search term`
- Function for searching analysis by is visible and search term
- Tests

### Changed

- Removed function analysis
- Removed Analysis assignment in basehandle

### Fixed

- Retrn object not query in `get_analyses_by_status_started_at_and_comment(`

### How to prepare for test
- [ ] ssh to hasta (depending on type of change)
- [ ] activate stage: `us`
- [ ] request trailblazer-stage on hasta: `paxa`
- [ ] install on stage:
    ```Shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_trailblazer -t trailblazer -b refactor-analyses -a
    ```
- [ ] ssh to clinical-db (depending on type of change)
- [ ] install on stage:
`bash /home/proj/production/servers/resources/clinical-db.scilifelab.se/update-trailblazer-ui-stage.sh [THIS-BRANCH-NAME]`

### How to test
- [ ] login to ...
- [ ] do ...

### Expected test outcome
- [ ] check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review
- [ ] tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a
- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

File: tests/store/filters/test_analysis_filters.py [hunk 1]
```diff
@@ -84,6 +86,136 @@ def test_filter_analyses_by_comment_when_not_matching(analysis_store: MockStore)
...
+    # THEN the analysis should match the original
+    assert existing_analysis == analyses.first()
+
+
+def test_filter_analyses_by_is_visible_when_false(analysis_store: MockStore):
+    """Test return analysis when is visible is false."""
+    # GIVEN a store containing analyses
+    existing_analysis: Analysis = analysis_store.get_query(table=Analysis).first()
+
+    # GIVEN a not visible analysis
```
Comment (ChrOertlin) [inline tests/store/filters/test_analysis_filters.py:114]:
```suggestion
    # GIVEN an invisible analysis
```

File: trailblazer/store/filters/analyses_filters.py [hunk 2]
```diff
@@ -23,6 +24,23 @@ def filter_analyses_by_entry_id(analyses: Query, analysis_id: int, **kwargs) ->
...
+
+def filter_analyses_by_search_term(analyses: Query, search_term: str, **kwargs) -> Query:
+    """Filter analyses by search term using multiple fields."""
+    return analyses.filter(
+        sqlalchemy.or_(
+            Analysis.family.ilike(f"%{search_term}%"),
+            Analysis.status.ilike(f"%{search_term}%"),
+            Analysis.data_analysis.ilike(f"%{search_term}%"),
+            Analysis.comment.ilike(f"%{search_term}%"),
+        )
```
Comment (ChrOertlin) [inline trailblazer/store/filters/analyses_filters.py:40]: Not sure if this function is a tad too general in terms of what the allowed fields are it searches.
What is the end goal of this filter? Is it used for display purposes only ?
What if we have an Analysis.comment that contains a `data_analysis` that is not the same as the `Analysis.data_analysis` one. Could this lead to issues?

**Review by ChrOertlin — APPROVED** [review]
Nice work! 👍

File: trailblazer/server/api.py [hunk 3]
```diff
@@ -54,18 +55,17 @@ def analyses():
     """Display analyses."""
     per_page = int(request.args.get("per_page", 100))
     page = int(request.args.get("page", 1))
-    query = store.analyses(
-        status=request.args.get("status"),
-        query=request.args.get("query"),
-        is_visible=request.args.get("is_visible") == "true" or None,
+    analyses: Query = store.get_analyses_query_by_search_term_and_is_visible(
+        search_term=request.args.get("query"),
+        is_visible=bool(request.args.get("is_visible")),
```
Comment (seallard) [inline trailblazer/server/api.py:60]: I'd place the extraction of the parameters above or in a separate function, matter of taste though. Feel free to disregard. Decreases the cognitive load a little to not have to understand the call and what the args are at once

**Review by seallard — APPROVED** [review]
Looks good :+1:

File: trailblazer/store/filters/analyses_filters.py (hunk 2 above)
Comment (henrikstranneheim) [reply trailblazer/store/filters/analyses_filters.py:40]: Agreed, but needs some work in the FE to remodel this. Should not be done now. This function is only used in the FE for display ad very greedy searching so no real harm.

**Comment by henrikstranneheim** [comment]
### CLI and API ✅ 

```
θ62° [henrik.stranneheim@hasta:~] [S_base] 1m12s $ trailblazer scan
Error updating analysis for: case - grandaphid : PermissionError
Error updating analysis for: case - goodbengal : MissingSqueueOutput
Error updating analysis for: case - helpedlab : MissingSqueueOutput
Error updating analysis for: case - newgopher : PermissionError
Error updating analysis for: case - onestag : PermissionError
Error updating analysis for: case - funnyaphid : FileNotFoundError
Error updating analysis for: case - proudfish : PermissionError
Error updating analysis for: case - setcolt : PermissionError
Error updating analysis for: case - handyshrimp : FileNotFoundError
Error updating analysis for: case - divinetomcat : PermissionError
Error updating analysis for: case - HW2V2DRX2 : FileNotFoundError
Error updating analysis for: case - savingstag : FileNotFoundError
Error updating analysis for: case - topewe : PermissionError
Error updating analysis for: case - fullgoat : PermissionError
Error updating analysis for: case - smoothtuna : FileNotFoundError
Error updating analysis for: case - cosmicslug : PermissionError
Error updating analysis for: case - readyhorse : PermissionError
Error updating analysis for: case - valueddog : FileNotFoundError
Error updating analysis for: case - lovingkoala : PermissionError
Error updating analysis for: case - topmarmot : PermissionError
Error updating analysis for: case - rarecow : PermissionError
Error updating analysis for: case - maturebeagle : PermissionError
Error updating analysis for: case - fithusky : PermissionError
Error updating analysis for: case - moralsnake : PermissionError
Error updating analysis for: case - drivenearwig : PermissionError
Error updating analysis for: case - nativetick : PermissionError
Error updating analysis for: case - smoothworm : PermissionError
Error updating analysis for: case - honestmoray : PermissionError
Error updating analysis for: case - poeticimpala : PermissionError
Error updating analysis for: case - mainsalmon : PermissionError
Error updating analysis for: case - knownloon : PermissionError
Error updating analysis for: case - merrybat : PermissionError
Error updating analysis for: case - quietdoe : PermissionError
Error updating analysis for: case - goldencolt : PermissionError
Error updating analysis for: case - aliveswine : PermissionError
Error updating analysis for: case - bigdragon : PermissionError
Error updating analysis for: case - fitburro : PermissionError
Error updating analysis for: case - uniquegrub : PermissionError
Error updating analysis for: case - fullnewt : PermissionError
Error updating analysis for: case - coolswine : PermissionError
Error updating analysis for: case - fleethyena : PermissionError
Error updating analysis for: case - smoothwombat : PermissionError
Error updating analysis for: case - coolcougar : PermissionError
Error updating analysis for: case - busytoad : PermissionError
Error updating analysis for: case - knownbull : PermissionError
Error updating analysis for: case - clearliger : PermissionError
Error updating analysis for: case - sunnybaboon : PermissionError
Error updating analysis for: case - flyingimp : PermissionError
Error updating analysis for: case - stableimp : PermissionError
Error updating analysis for: case - goodrobin : PermissionError
Error updating analysis for: case - comicmite : PermissionError
Error updating analysis for: case - rapidnewt : PermissionError
Error updating analysis for: case - valuedlab : PermissionError
Error updating analysis for: case - secureghoul : PermissionError
Error updating analysis for: case - hipgull : PermissionError
Error updating analysis for: case - livingmoth : PermissionError
Error updating analysis for: case - puredrum : PermissionError
Error updating analysis for: case - grandkite : PermissionError
Error updating analysis for: case - soundminnow : PermissionError
Error updating analysis for: case - cosmicsponge : PermissionError
Error updating analysis for: case - holydory : PermissionError
Error updating analysis for: case - solidpoodle : PermissionError
Error updating analysis for: case - warmghoul : PermissionError
Error updating analysis for: case - fondbengal : PermissionError
Error updating analysis for: case - awaregull : PermissionError
Error updating analysis for: case - onlocust : PermissionError
Error updating analysis for: case - modelrhino : PermissionError
Error updating analysis for: case - intentalien : PermissionError
Error updating analysis for: case - notedbat : PermissionError
Error updating analysis for: case - nobleflea : PermissionError
Error updating analysis for: case - soughtcub : PermissionError
Error updating analysis for: case - stablekite : PermissionError
Error updating analysis for: case - holyfowl : PermissionError
Error updating analysis for: case - starmacaw : PermissionError
Error updating analysis for: case - toughcoral : PermissionError
Error updating analysis for: case - neededhawk : PermissionError
Error updating analysis for: case - causaldonkey : PermissionError
Error updating analysis for: case - movingcivet : PermissionError
Error updating analysis for: case - nextwalrus : PermissionError
Error updating analysis for: case - warmcivet : PermissionError
Error updating analysis for: case - steadypup : PermissionError
Error updating analysis for: case - divinesnake : PermissionError
Error updating analysis for: case - bossmidge : PermissionError
Error updating analysis for: case - savedsnipe : PermissionError
Error updating analysis for: case - oncorgi : PermissionError
Error updating analysis for: case - largeliger : PermissionError
Error updating analysis for: case - alertox : PermissionError
Error updating analysis for: case - suitedelk : PermissionError
Error updating analysis for: case - hardycivet : PermissionError
Error updating analysis for: case - epicfalcon : PermissionError
Error updating analysis for: case - growntapir : PermissionError
Error updating analysis for: case - rightmonkey : PermissionError
Error updating analysis for: case - rightwhale : PermissionError
Error updating analysis for: case - valuedgibbon : PermissionError
Error updating analysis for: case - nativegannet : PermissionError
Error updating analysis for: case - richwombat : PermissionError
Error updating analysis for: case - firmcoral : PermissionError
Error updating analysis for: case - grandhare : PermissionError
Error updating analysis for: case - wiredkodiak : PermissionError
Error updating analysis for: case - simplerat : PermissionError
Error updating analysis for: case - brightmayfly : PermissionError
Error updating analysis for: case - pumpedhyena : PermissionError
Error updating analysis for: case - tidyant : PermissionError
Error updating analysis for: case - firstcod : PermissionError
Error updating analysis for: case - mutualjoey : PermissionError
Error updating analysis for: case - tidyshrew : PermissionError
Error updating analysis for: case - strongprawn : PermissionError
Error updating analysis for: case - holycicada : PermissionError
Error updating analysis for: case - restedmako : PermissionError
Error updating analysis for: case - normalfish : PermissionError
Error updating analysis for: case - busywren : PermissionError
Error updating analysis for: case - fleetturtle : PermissionError
Error updating analysis for: case - freshoriole : PermissionError
Error updating analysis for: case - directamoeba : PermissionError
Error updating analysis for: case - gameoyster : PermissionError
Error updating analysis for: case - setracer : PermissionError
Error updating analysis for: case - gentlemolly : PermissionError
Error updating analysis for: case - vastswift : PermissionError
Error updating analysis for: case - upwardperch : PermissionError
Error updating analysis for: case - divinewolf : PermissionError
Error updating analysis for: case - quickcougar : PermissionError
Error updating analysis for: case - actualkid : PermissionError
Error updating analysis for: case - grandnewt : PermissionError
Error updating analysis for: case - caringmullet : PermissionError
Error updating analysis for: case - sacredpoodle : PermissionError
Error updating analysis for: case - funnylouse : PermissionError
Error updating analysis for: case - sharplark : PermissionError
Error updating analysis for: case - mastercattle : PermissionError
Error updating analysis for: case - freshhound : PermissionError
Error updating analysis for: case - quietmutt : PermissionError
Error updating analysis for: case - guidedmutt : PermissionError
Error updating analysis for: case - fithawk : PermissionError
Error updating analysis for: case - modelhornet : PermissionError
Error updating analysis for: case - wholekodiak : PermissionError
Error updating analysis for: case - kindlab : PermissionError
Error updating analysis for: case - freegator : PermissionError
Error updating analysis for: case - modesteft : PermissionError
Error updating analysis for: case - proudwalrus : PermissionError
Error updating analysis for: case - minttroll : PermissionError
Error updating analysis for: case - manycobra : PermissionError
Error updating analysis for: case - gamesawfly : PermissionError
Error updating analysis for: case - holysatyr : PermissionError
Error updating analysis for: case - politecobra : PermissionError
Error updating analysis for: case - livingtahr : PermissionError
Error updating analysis for: case - solidclam : PermissionError
Error updating analysis for: case - cleverbaboon : PermissionError
Error updating analysis for: case - propercat : PermissionError
Error updating analysis for: case - modesthare : PermissionError
Error updating analysis for: case - primepigeon : PermissionError
Error updating analysis for: case - gladshark : PermissionError
Error updating analysis for: case - suitedpiglet : PermissionError
Error updating analysis for: case - eagerzebra : PermissionError
Error updating analysis for: case - divinewalrus : FileNotFoundError
slurm_load_jobs error: Invalid job id specified
Error updating analysis for: case - wholecivet_rsync : CalledProcessError
Failed to update fluentgull - 279455: FileNotFoundError
Status in Tower: hardygrouse - 279457 - 2L5JgBVq71Zv19
slurm_load_jobs error: Invalid job id specified
Error updating analysis for: case - wholecivet_rsync : CalledProcessError
Error updating analysis for: case - setamoeba : FileNotFoundError
Error updating analysis for: case - acetuna : FileNotFoundError
All analyses updated!
```

Screenshot 2023-11-07 at 08 42 04


Screenshot 2023-11-07 at 08 42 18

Screenshot 2023-11-07 at 08 42 49

**Comment by henrikstranneheim** [comment]
Deployed to production:

```
Log deploy... done.
trailblazer, version 19.0.1
```

**Pull request merged**